app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb

```diff
@@ -474,7 +474,21 @@ def client_authorized?
     def update_host_registered_through(host, headers)
       parent_host = get_parent_host(headers)
       host.subscription_facet.update_attribute(:registered_through, parent_host)
-      content_source_id = get_content_source_id(parent_host)
+      set_host_content_source(host, parent_host)
+    end
+
+    def registering_thru_load_balancer?(hostname)
+      # If the request doesn't come from any known smart proxy, it came from a load balancer.
+      get_content_source_id(hostname).nil?
```

Comment:
Is this a safe assumption? Could you instead check all smart-proxies registration URLs and see if they match the hostname?

Reply:
Yeah, that would be better. At this point I should probably just change it not to use the header, too..

Reply:
Updated to check if hostname is a known load balancer. :+1: